### ISSUE TITLE
Unable to handle test generation helper correctly

### ISSUE THREAD

**Issue by xlc** [oven-sh/bun#7873]
### What version of Bun is running?

1.0.20+09d51486e

### What platform is your computer?

Darwin 23.1.0 arm64 arm

### What steps can reproduce the bug?

Clone repo https://github.com/xlc/bun-test-issue
`bun test`

### What is the expected behavior?

Two tests pass, one for a.test.ts and one for b.test.ts
Two snapshot files generated, one for a.test.ts and one for b.test.ts

### What do you see instead?

```
bun test v1.0.20 (09d51486)

tests/b.test.ts:
✓ b [2.33ms]

tests/a.test.ts:

 1 pass
 0 fail
 1 snapshots, 1 expect() calls
Ran 1 tests across 2 files. [84.00ms]
```

only 1 test run and only 1 snapshot file generated under helper.ts.snap

### Additional information

I am migrating some tests from vitest to bun and this is a blocking issue for the migration.

Basically bun test unable to correctly handle test generated within a helper function

**Comment by xlc** [comment]
I spent a bit time trying to see if I can fix this, but seems like a relatively large change is required.

From my very limited understanding (this is me first time reading zig code):

- bun is rewriting `import * from 'bun:test'` to `globalThis.Bun.jest(import.meta.path)` https://github.com/oven-sh/bun/blob/f02ef8fb35664cfdced23c35237b168ec802b2f5/src/js_printer.zig#L1027
- bun.jest is called with the filename of the import, instead of the invoke so `tests/helper.ts` is used as the test filename, instead of `tests/a.test.ts` or `tests/b.test.ts`
- The filename here is correct, so the solution is somehow pass it, instead of using the bun.jest hack https://github.com/oven-sh/bun/blob/f02ef8fb35664cfdced23c35237b168ec802b2f5/src/cli/test_command.zig#L979

**Comment by xlc** [comment]
Knowing how it worked now, I can workaround this by doing `const { beforeEach, describe, it } = (Bun as any).jest(filename)` and manually passing `import.meta.path` from the test source to the helper
